proof/courier.go

```diff
@@ -439,6 +439,17 @@ func (h *HashMailBox) RecvAck(ctx context.Context, sid streamID) error {
 		return fmt.Errorf("unable to create read stream: %w", err)
 	}
 
+	// Ensure the read stream is closed before returning.
+	defer func() {
+		log.Debugf("Closing hashmail read stream (sid=%x)", sid[:])
+
+		err := readStream.CloseSend()
```

Comment:
I don't think this is necessary. If you look into `*hashMailClient.RecvStream()`, you see that `x.ClientStream.CloseSend()` is already called. So I think this might even lead to an error?

Reply:
Dropped last commit in latest push.